# Plot Pie Charts to Visualize the proportion of people have strokes

In [1]:
# import package
import pandas as pd 
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [2]:
df = pd.read_csv('healthcare-dataset-stroke-data.csv')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
id                   5110 non-null int64
gender               5110 non-null object
age                  5110 non-null float64
hypertension         5110 non-null int64
heart_disease        5110 non-null int64
ever_married         5110 non-null object
work_type            5110 non-null object
Residence_type       5110 non-null object
avg_glucose_level    5110 non-null float64
bmi                  4909 non-null float64
smoking_status       5110 non-null object
stroke               5110 non-null int64
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


5110 entries, 12 fields. Will plot pie charts to show proportion of people had strokes among the group of smoking_status

In [4]:
df.smoking_status.value_counts()

never smoked       1892
Unknown            1544
formerly smoked     885
smokes              789
Name: smoking_status, dtype: int64

In [5]:
ns = df.query("smoking_status == 'never smoked'").stroke.value_counts().to_list()
uk = df.query("smoking_status == 'Unknown'").stroke.value_counts().to_list()
fs = df.query("smoking_status == 'formerly smoked'").stroke.value_counts().to_list()
sm = df.query("smoking_status == 'smokes'").stroke.value_counts().to_list()
label = ['yes', 'no']

In [6]:
fig = make_subplots(rows=2, cols=2,
    specs=[[{"type": "pie"}, {"type": "pie"}],
           [{"type": "pie"}, {"type": "pie"}]],
           subplot_titles=("Never Smoked", "Unknown", "Formerly Smoked", "Smokes"))

fig.add_trace(
    go.Pie(labels = label, values = ns, pull=[0.2,0]), 
    row=1, col=1
)
fig.add_trace(
    go.Pie(labels = label, values = uk, pull=[0.2,0]), 
    row=1, col=2
)
fig.add_trace(
    go.Pie(labels = label, values = fs, pull=[0.2,0]), 
    row=2, col=1
)
fig.add_trace(
    go.Pie(labels = label, values = sm, pull=[0.2,0]), 
    row=2, col=2
)
fig.update_layout(title_text="% of Stroke among Smoking Status")
fig.show()